Cython/Compiler/Nodes.py

```diff
@@ -8569,9 +8569,6 @@ def analyse_declarations(self, env):
         if not env.is_module_scope:
             error(self.pos, "cimport only allowed at module level")
             return
-        if self.relative_level and self.relative_level > env.qualified_name.count('.'):
-            error(self.pos, "relative cimport beyond main package is not allowed")
-            return
```

Comment:
This part was removed, in order to fix failing TC introduced in: https://github.com/cython/cython/pull/4552/commits/bd361e215277f81d1ac358269adf34879007e804

Reply:
I have fixed the condition and introduced back this error.